mixer-ngx/src/ngx/server_config.rs

```diff
@@ -1,41 +1,90 @@
 extern crate ngx_rust;
 
-use ngx_rust::bindings:: { ngx_uint_t, ngx_str_t } ;
+use std::collections::HashMap;
+use ngx_rust::bindings:: { ngx_uint_t, ngx_str_t, ngx_array_t } ;
 
 use ngx_mixer_transport::attribute::attr_wrapper::AttributeWrapper;
 use ngx_mixer_transport::attribute::global_dict::{ SOURCE_IP, SOURCE_UID, SOURCE_SERVICE, SOURCE_PORT,
-                    DESTINATION_SERVICE,DESTINATION_IP,DESTINATION_UID
+                    SOURCE_LABELS, DESTINATION_SERVICE, DESTINATION_IP, DESTINATION_UID, DESTINATION_LABELS
 };
 
 use super::config::MixerConfig;
+use std::net::Ipv4Addr;
 
+#[repr(C)]
+pub struct service_labels {
+    pub key:                 *const ngx_array_t,
+    pub value:               *const ngx_array_t
+
+}
+
+// ngx_http_mixer_srv_conf_t struct made compatible with data type in c module using #[repr(C)]
 #[repr(C)]
 pub struct ngx_http_mixer_srv_conf_t {
 
     pub destination_service:    ngx_str_t,
     pub destination_uid:        ngx_str_t,
     pub destination_ip:         ngx_str_t,
+    pub destination_labels:     service_labels,
     pub source_ip:              ngx_str_t,
     pub source_uid:             ngx_str_t,
     pub source_service:         ngx_str_t,
-    pub source_port:            ngx_uint_t
+    pub source_port:            ngx_uint_t,
+    pub source_labels:          service_labels
 }
 
 impl MixerConfig for  ngx_http_mixer_srv_conf_t  {
 
     fn process_istio_attr(&self, attr: &mut AttributeWrapper) {
+        ngx_event_debug!("Processing server attributes");
 
+        ngx_event_debug!("{}: {}",DESTINATION_SERVICE, self.destination_service.to_str());
```

Comment:
this debugging makes the code hard to read. I would suggest grouping this together below or above the other code. `to_str()` is called two times for every variable in this approach too.